pkg/controller/service/service.go

```diff
@@ -13,25 +16,49 @@ func RenderServices(req router.Request, resp router.Response) error {
 	// reset, should get repopulated
 	svcInstance.Status.Endpoints = nil
 
-	objs, _, err := services.ToK8sService(req, svcInstance)
-	if err != nil {
-		return err
+	var svcList []*v1.ServiceInstance
+	http2Ports := ports.ByProtocol(svcInstance.Spec.Ports, true, v1.ProtocolHTTP2)
+	otherPorts := ports.ByProtocol(svcInstance.Spec.Ports, false, v1.ProtocolHTTP2)
+	if len(http2Ports) > 0 {
+		svcCopy := svcInstance.DeepCopy()
+		svcCopy.Spec.Ports = http2Ports
+		svcCopy.Name = fmt.Sprintf("%s-%s", svcInstance.Name, v1.ProtocolHTTP2)
+		if svcCopy.Spec.Annotations == nil {
+			svcCopy.Spec.Annotations = map[string]string{}
+		}
+		svcCopy.Spec.Annotations["traefik.ingress.kubernetes.io/service.serversscheme"] = "h2c"
+		svcList = append(svcList, svcCopy)
```

Comment:
Do you have to actually create svcInstance or can we just create separate services with special annotations? Seems like we should not be creating svcinstances here.

Reply:
So I wonder if you should be able to create separate corev1.service in `services.ToK8sService(req, svc)`. So for http2 port it will be a separate service with special annotation

Reply:
Discussed offline, this is doing the right thing to create separate service by tweaking serviceInstance that is passed to `ToK8sService`